src/python/py_rag.cpp

```diff
@@ -0,0 +1,139 @@
+                "texts ",
+                "Computes embeddings for a text")
+            .def(
+                "start_embed_query_async",
+                [](TextEmbeddingPipeline& pipe, std::string& text) -> void {
+                    py::gil_scoped_release rel;
+                    pipe.start_embed_query_async(text);
+                },
+                py::arg("texts"),
+                "texts ",
+                "Asynchronously computes embeddings for a text")
+            .def(
+                "wait_embed_query",
+                [](TextEmbeddingPipeline& pipe) -> py::typing::Union<EmbeddingResult> {
+                    EmbeddingResult res;
+                    {
+                        py::gil_scoped_release rel;
+                        res = pipe.wait_embed_query();
+                    }
+                    return py::cast(res);
+                },
+                "Waits computed embeddings for a text");
+
+    py::enum_<TextEmbeddingPipeline::PoolingType>(text_embedding_pipeline, "PoolingType")
+        .value("CLS", TextEmbeddingPipeline::PoolingType::CLS)
+        .value("MEAN", TextEmbeddingPipeline::PoolingType::MEAN);
+
+    py::class_<TextEmbeddingPipeline::Config>(text_embedding_pipeline,
+                                              "Config",
+                                              "This class is used for storing TextEmbeddingPipeline config.")
+        .def(py::init<>())
+        .def(py::init([](py::kwargs kwargs) {
+            return TextEmbeddingPipeline::Config(pyutils::kwargs_to_any_map(kwargs));
+        }))
+        .def_readwrite("max_length", &TextEmbeddingPipeline::Config::max_length)
+        .def_readwrite("pooling_type", &TextEmbeddingPipeline::Config::pooling_type)
+        .def_readwrite("normalize", &TextEmbeddingPipeline::Config::normalize)
+        .def_readwrite("query_instruction", &TextEmbeddingPipeline::Config::query_instruction)
+        .def_readwrite("embed_instruction", &TextEmbeddingPipeline::Config::embed_instruction);
+
+    text_embedding_pipeline.def(
+        py::init([](const std::filesystem::path& models_path,
+                    const std::string& device,
+                    const std::optional<TextEmbeddingPipeline::Config>& config,
+                    const py::kwargs& kwargs) {
+            ScopedVar env_manager(pyutils::ov_tokenizers_module_path());
+
+            if (config.has_value()) {
+                return std::make_unique<TextEmbeddingPipeline>(models_path,
+                                                               device,
+                                                               *config,
+                                                               pyutils::kwargs_to_any_map(kwargs));
+            }
+            return std::make_unique<TextEmbeddingPipeline>(models_path, device, pyutils::kwargs_to_any_map(kwargs));
+        }),
+        py::arg("models_path"),
+        "folder with openvino_model.xml and openvino_tokenizer[detokenizer].xml files",
```

Comment:
```suggestion
        "folder with openvino_model.xml and openvino_tokenizer.xml files",
```
Do you care about detokenizer?
Also add this to C++ docs

Reply:
Fixed